portability/toku_instrumentation.h

```diff
@@ -0,0 +1,47 @@
+/* -*- mode: C++; c-basic-offset: 4; indent-tabs-mode: nil -*- */
+// vim: ft=cpp:expandtab:ts=8:sw=4:softtabstop=4:
+#ident "$Id$"
+/*======
+  This file is part of PerconaFT.
+
+
+  Copyright (c) 2006, 2015, Percona and/or its affiliates. All rights reserved.
+
+  PerconaFT is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License, version 2,
+  as published by the Free Software Foundation.
+
+  PerconaFT is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with PerconaFT.  If not, see <http://www.gnu.org/licenses/>.
+
+  ----------------------------------------
+
+  PerconaFT is free software: you can redistribute it and/or modify
+  it under the terms of the GNU Affero General Public License, version 3,
+  as published by the Free Software Foundation.
+
+  PerconaFT is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU Affero General Public License for more details.
+
+  You should have received a copy of the GNU Affero General Public License
+  along with PerconaFT.  If not, see <http://www.gnu.org/licenses/>.
+  ======= */
+
+#ident "Copyright (c) 2006, 2015, Percona and/or its affiliates. All rights reserved."
+
+#pragma once
+
+#include <stdio.h> // FILE
+
+struct TOKU_FILE
```

Comment:
I was hoping that the file containing the TOKU_FILE struct/class would be in a similarly named file.  One reason:  consumers of this object should not know anything about whether it is instrumented or not.  These consumers would just have a dependency on something like toku_file.h.  This is especially true if we add methods to the object and it needs a corresponding .cpp/.cc file for the implementation.

That said...
- I understand that the design of the instrumentation will be optionally compiled away, so dependents can't access instrumentation details without compiler errors when the fields are not there. 
- The instrumentation for a file needs to live somewhere, and this is as a logical place as any.  
- I also recognize the possible redundancy of having a file named toku_file.h and then a separate file header with the compiled in instrumentation.